packages/medusa/src/strategies/batch-jobs/product/export.ts

```diff
@@ -0,0 +1,387 @@
+    this.fileService_ = fileService
+    this.productRepository_ = productRepository
+  }
+
+  async buildTemplate(): Promise<string> {
+    return this.buildHeader()
+  }
+
+  async prepareBatchJobForProcessing(
+    batchJob: AdminPostBatchesReq,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    req: Express.Request
+  ): Promise<AdminPostBatchesReq> {
+    const {
+      limit,
+      offset,
+      order,
+      fields,
+      expand,
+      filterable_fields,
+      ...context
+    } = batchJob.context
+
+    const listConfig = prepareListQuery(
+      {
+        limit,
+        offset,
+        order,
+        fields,
+        expand,
+      },
+      {
+        isList: true,
+        defaultRelations: this.defaultRelations_,
+      }
+    )
+
+    batchJob.context = {
+      ...(context ?? {}),
+      list_config: listConfig,
+      filterable_fields,
+    }
+
+    return batchJob
+  }
+
+  async processJob(batchJobId: string): Promise<void> {
+    let offset = 0
+    let limit = this.BATCH_SIZE
+    let advancementCount = 0
+    let productCount = 0
+
+    return await this.atomicPhase_(
+      async (transactionManager) => {
+        let batchJob = await this.batchJobService_
+          .withTransaction(transactionManager)
+          .retrieve(batchJobId)
+
+        const { writeStream, fileKey, promise } =
+          await this.fileService_.getUploadStreamDescriptor({
+            name: "product-export",
+            ext: "csv",
+          })
+
+        const header = await this.buildHeader()
+        writeStream.write(header)
+
+        offset = batchJob.context.list_config?.skip ?? offset
+        limit = batchJob.context.list_config?.take ?? this.BATCH_SIZE
+        advancementCount = batchJob.result.advancement_count ?? advancementCount
+
+        const { list_config = {}, filterable_fields = {} } = batchJob.context
+        const [productList, count] = await this.productService_
+          .withTransaction(transactionManager)
+          .listAndCount(filterable_fields, {
+            ...list_config,
+            skip: offset,
+            take: limit,
+          } as FindProductConfig)
+
+        productCount = count
+        let products: Product[] = productList
+
+        while (offset < productCount) {
+          if (!products?.length) {
+            products = await this.productService_
+              .withTransaction(transactionManager)
+              .list(filterable_fields, {
+                ...list_config,
+                skip: offset,
+                take: limit,
+              } as FindProductConfig)
+          }
+
+          products.forEach((product: Product) => {
+            const lines = this.buildProductVariantLines(product)
+            lines.forEach((line) => writeStream.write(line))
+          })
+
+          advancementCount += products.length
+          offset += products.length
+          products.length = 0
+
+          batchJob = await this.batchJobService_.update(batchJobId, {
```

Comment:
```suggestion
          batchJob = await this.batchJobService_.withTransaction(transactionManager).update(batchJobId, {
```

Reply:
ofc, just an issue I ran into with the orderExporter, though I can't remember if this was the instance of update, so I just threw the comment here also for future reference